generic/tests/vlan.py

```diff
@@ -91,11 +90,21 @@ def rem_vlan(test, session, v_id, iface="eth0", cmd_type="ip"):
                               test.log.info)
         return session.cmd_status(rem_vlan_cmd)
 
+    def find_free_port(src):
+        """
+        Returns a free port on the src guest
+        """
+        check_cmd = "netstat -nultp |awk '(NR>2){print $4}' | awk -F':' '{print $NF}'"
+        used_port_names = set(sessions[src].cmd_output(check_cmd).strip().splitlines())
+        port_range = list(range(1025, 65535))
+        free_ports = list(set(port_range).difference(set(used_port_names)))
+        return free_ports[0]
+
     def nc_transfer(test, src, dst):
         """
         Transfer file by netcat
         """
-        nc_port = utils_misc.find_free_port(1025, 5334, vm_ip[dst])
+        nc_port = find_free_port(src)
```

Comment:
You can see `utils_misc.find_free_port(1025, 5334, vm_ip[dst])`, **dst**, so the nc_port is a free port in the dst guest.

Reply:
Thanks for your reminder, I have always misunderstood it as the ip address in the nc command. `src` has been replaced to `dst`, please help review it again, thanks a lot.